### ISSUE TITLE
Wrong offset calculations in lane widths for junctions with multiple lanes 

### ISSUE THREAD

**Issue by fdtmbartelt** [pyoscx/scenariogeneration#171]
By testing the CommonJunctionCreator I found an issue while using multiple lanes with different widths. 
![grafik](https://github.com/pyoscx/scenariogeneration/assets/101714080/0a3f69c5-b3fd-4b42-beb2-607b14b7ebc0)
The connection with only one lane works perfect, but if I added an additional lane the calculation for the lane offset seems to be incorrect, because the wrong width is used.
![grafik](https://github.com/pyoscx/scenariogeneration/assets/101714080/2722d3ad-e336-44c8-89ab-265a13f6afb5)
For testing this issue I upload the following python file as txt 
[Complex_Junctiontest.txt](https://github.com/pyoscx/scenariogeneration/files/12080893/Complex_Junctiontest.txt)
It would be very nice if someone could confirm the behavior or describe a workaround that fixes the issue.

**Comment by MandolinMicke** [comment]
Yes, this is a known limitation that I would like to fix, but it will require a big refactorization, hence I haven't done it yet..

**Comment by mander76** [comment]
There was a pull request about this recently, https://github.com/pyoscx/scenariogeneration/pull/191 (not released yet on pypi)
Could you try if that solves your problem aswell?

**Comment by MandolinMicke** [comment]
Can you confirm that the latest updates has fixed your issue?
